File: backend/ai_server/ai_server/pkg/utils/connection_url_builder.py
import os

def build_connection_url(name: str) -> str:
    if name == "postgres":
        return (
            f"postgresql://{os.getenv('DATABASE_USER')}:{os.getenv('DATABASE_PASSWORD')}"
            f"@{os.getenv('DATABASE_HOST')}:{os.getenv('DATABASE_PORT')}/{os.getenv('DATABASE_NAME')}?"
            f"sslmode={os.getenv('DATABASE_SSL_MODE', 'prefer')}"
        )
    elif name == "mysql":
        return (
            f"mysql+mysqlconnector://{os.getenv('DATABASE_USER')}:{os.getenv('DATABASE_PASSWORD')}"
            f"@{os.getenv('DATABASE_HOST')}:{os.getenv('DATABASE_PORT')}/{os.getenv('DATABASE_NAME')}"
        )
    elif name == "redis":
        return (
            f"redis://{os.getenv('REDIS_USERNAME')}:{os.getenv('REDIS_PASSWORD')}"
            f"@{os.getenv('REDIS_HOST')}:{os.getenv('REDIS_PORT')}/{os.getenv('REDIS_DB')}"
        )
    elif name == "fastapi":
        return f"http://{os.getenv('SERVER_HOST')}:{os.getenv('SERVER_PORT')}"
    elif name == "kafka":
        bootstrap_servers = os.getenv("KAFKA_BOOTSTRAP_SERVERS", "localhost:9092")
        topics = os.getenv("KAFKA_TOPICS", "topic1,topic2")
        return f"kafka://{bootstrap_servers}?topics={topics}"
    elif name == "spark":
        master = os.getenv("SPARK_MASTER", "local[*]")
        app_name = os.getenv("SPARK_APP_NAME", "MySparkApp")
        return f"spark://{master}?appName={app_name}"
    else:
        raise ValueError(f"Connection name '{name}' is not supported.")

File: backend/api_server/pkg/utils/connection/connection_url_builder.go
package connection

import (
	"fmt"
	"os"
)

func ConnectionURLBuilder(n string) (string, error) {
	var url string

	switch n {
	case "postgres":
		url = fmt.Sprintf(
			"host=%s port=%s user=%s password=%s dbname=%s sslmode=%s",
			os.Getenv("DATABASE_HOST"),
			os.Getenv("DATABASE_PORT"),
			os.Getenv("DATABASE_USER"),
			os.Getenv("DATABASE_PASSWORD"),
			os.Getenv("DATABASE_NAME"),
			os.Getenv("DATABASE_SSL_MODE"),
		)
	case "mysql":
		url = fmt.Sprintf(
			"%s:%s@tcp(%s:%s)/%s",
			os.Getenv("DATABASE_USER"),
			os.Getenv("DATABASE_PASSWORD"),
			os.Getenv("DATABASE_HOST"),
			os.Getenv("DATABASE_PORT"),
			os.Getenv("DATABASE_NAME"),
		)
	case "redis":
		url = fmt.Sprintf(
			"%s:%s",
			os.Getenv("REDIS_HOST"),
			os.Getenv("REDIS_PORT"),
		)
	case "fiber":
		url = fmt.Sprintf(
			"%s:%s",
			os.Getenv("SERVER_HOST"),
			os.Getenv("SERVER_PORT"),
		)
	case "kafka":
		url = fmt.Sprintf(
			"%s:%s",
			os.Getenv("KAFKA_HOST"),
			os.Getenv("KAFKA_PORT"),
		)
	case "mongodb_interaction":
		url = fmt.Sprintf(
			"mongodb://%s:%s@%s:%s/%s",
			os.Getenv("INTERACTIONS_DB_USER"),
			os.Getenv("INTERACTIONS_DB_PASSWORD"),
			os.Getenv("INTERACTIONS_DB_HOST"),
			os.Getenv("INTERACTIONS_DB_PORT"),
			os.Getenv("INTERACTIONS_DB_NAME"),
		)
	default:
		return "", fmt.Errorf("connection name '%v' is not supported", n)
	}

	return url, nil
}
